horizon/data_manager/policy_store.py

```diff
@@ -0,0 +1,139 @@
+        policy_updater_enabled: bool = True,
+        cache_policy_data: bool = False,
+        tls_client_cert: Optional[str] = None,
+        tls_client_key: Optional[str] = None,
+        tls_ca: Optional[str] = None,
+    ):
+        super().__init__(
+            opa_server_url,
+            opa_auth_token,
+            auth_type,
+            oauth_client_id,
+            oauth_client_secret,
+            oauth_server,
+            data_updater_enabled,
+            policy_updater_enabled,
+            cache_policy_data,
+            tls_client_cert,
+            tls_client_key,
+            tls_ca,
+        )
+        self._client = data_manager_client
+
+    async def set_policy_data(
+        self,
+        policy_data: JsonableValue,
+        path: str = "",
+        transaction_id: Optional[str] = None,
+    ):
+        parts = path.lstrip("/").split("/")
+        match parts:
+            case ["relationship_tuples", subject]:
+                for full_relation, targets in policy_data.items():
+                    relation = full_relation.lstrip("relation:")
+                    for target_type, target_objects in targets.items():
+                        for target in target_objects:
+                            # TODO missing subject_type and target_type
+                            await self._insert_fact(
+                                "relationships",
+                                {
+                                    "subject": subject,
+                                    "relation": relation,
+                                    "object": target,
+                                    "tenant": "",  # TODO unnecessary?
+                                },
+                            )
+            case ["role_assignments", full_user_key]:
+                user_key = full_user_key.lstrip("user:")
+                for subject, roles in policy_data.items():
+                    subject_type, subject_key = subject.split(":", 1)
+                    for role_key in roles:
+                        if subject_key == "__tenant":
+                            await self._insert_fact(
+                                "role_assignments",
+                                {
+                                    "actor": user_key,
+                                    "tenant": subject_key,
+                                    "role": role_key,
+                                    "resource": "",
+                                },
+                            )
+                        else:
+                            # TODO missing resource_type
+                            await self._insert_fact(
+                                "role_assignments",
+                                {
+                                    "actor": user_key,
+                                    "tenant": "",
+                                    "role": role_key,
+                                    "resource": subject_key,
```

Comment:
IIRC the resource should be the entire subject - <id>:<type> like "document:budget"